### ISSUE TITLE
Call still active if caller hands up before connected to volunteer

### ISSUE THREAD

**Issue by radius314** [radius314/yap#82]
**Describe the bug**
If a caller hangs up before connecting to a volunteer, when the volunteer joins they will be added to the call, but just hear music on hold indefinately.  

**To Reproduce**
1. Call in and enter parameters to find someone to talk to.
2. Wait until the call starts ringing the volunteer.
3. Caller hangs up.
4. Volunteer answers, presses 1 to accept the call.
5. Volunteer hears music on hold indefinately.

**Expected behavior**
If the call ends by the caller, the dial pattern should end as well, or relay some state that the caller end abruptly.

**Comment by radius314** [comment]
The hangup has to happen before the volunteer receives the call.

**Comment by radius314** [comment]
Fix is in unstable branch now
